include/urg_node/urg_node_driver.hpp

```diff
@@ -35,25 +35,34 @@
 #ifndef URG_NODE_URG_NODE_DRIVER_H
 #define URG_NODE_URG_NODE_DRIVER_H
 
+#include <chrono>
 #include <string>
-#include <ros/ros.h>
-#include <dynamic_reconfigure/server.h>
+#include <rclcpp/rclcpp.hpp>
+//#include <dynamic_reconfigure/server.h>
 #include <laser_proc/LaserTransport.h>
-#include <diagnostic_updater/diagnostic_updater.h>
-#include <diagnostic_updater/publisher.h>
-#include <urg_node/URGConfig.h>
-#include <std_srvs/Trigger.h>
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <diagnostic_updater/publisher.hpp>
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+//#include <urg_node/URGConfig.h>
+#include <std_srvs/srv/trigger.hpp>
 
-#include "urg_node/urg_c_wrapper.h"
+#include "urg_node_msgs/msg/status.hpp"
+#include "urg_node/urg_c_wrapper.hpp"
+
+//#include <tf2/tf.h>  // tf header for resolving tf prefix
+#include <rclcpp/parameter_client.hpp>
+#include <rcl_interfaces/msg/parameter.hpp>
+#include <rcl_interfaces/msg/set_parameters_result.hpp>
+#include <iostream>
 
 namespace urg_node
 {
-class UrgNode
+class UrgNode : public rclcpp::Node
 {
 public:
-  UrgNode();
-  UrgNode(ros::NodeHandle nh, ros::NodeHandle private_nh);
-  ~UrgNode();
+  explicit UrgNode(const std::string & node_name);
```

Comment:
If you have this also take a `const rclcpp::NodeOptions & options` parameter, it probably isn't too much further work to make this composable.

Reply:
I'd like to postpone this for a follow PR. I'd like to reformat the nodes later on to be a lifecycle node and then in the same stroke make them composable.